helpers/ultimoMovimientoDetalle.go

```diff
@@ -0,0 +1,140 @@
+						"err":     err,
+						"status":  "502",
+					}
+					return models.MovimientoDetalle{}, outputError
+				} else {
+					filtroJsonB = map[string]interface{}{
+						"RubroId": datosCuenta.RubroId,
+						// "FuenteFinanciamientoId": datosCuenta.FuenteFinanciamientoId,
+						"ActividadId": actividadInt,
+					}
+				}
+			}
+		}
+	} else if datosCuenta.ActividadId == "" {
+		if datosCuenta.RubroId != "" {
+			if datosCuenta.FuenteFinanciamientoId != "" {
+				if fuenteInt, err := strconv.Atoi(datosCuenta.FuenteFinanciamientoId); err != nil {
+					logs.Error(err)
+					outputError = map[string]interface{}{
+						"funcion": "GetUltimo - strconv.Atoi(datosCuenta.FuenteFinanciamientoId)",
+						"err":     err,
+						"status":  "502",
+					}
+					return models.MovimientoDetalle{}, outputError
+				} else {
+					filtroJsonB = map[string]interface{}{
+						"RubroId":                datosCuenta.RubroId,
+						"FuenteFinanciamientoId": fuenteInt,
+					}
+				}
+			}
+		}
+	}
+
+	data, _ := json.Marshal(filtroJsonB)
+
+	datosMovProcExterno := cuentaMovimientoDetalle.Mov_Proc_Ext
+
+	var query map[string]string
+
+	query = map[string]string{
+		"Detalle__json_contains":         string(data),
+		"MovimientoProcesoExternoId__Id": datosMovProcExterno,
+	}
+
+	// Se sugiere ordenar por fecha de modificación
+	sortby := []string{
+		"FechaModificacion",
+	}
+
+	// El orden descendente velará por traer el último registro modificado
+	order := []string{
+		"desc",
+	}
+
+	// Para traer el último
+	limit := int64(1)
+
+	// Nota: Se envían los parámetros de fields como nil y offset por default como 0
+	if result, err := models.GetAllMovimientoDetalle(query, nil, sortby, order, 0, limit); err != nil {
+		logs.Error(err)
+		outputError = map[string]interface{}{
+			"funcion": "GetUltimo - models.GetAllMovimientoDetalle(query, nil, sortby, order, 0, limit)",
+			"err":     err,
+			"status":  "502",
+		}
+		return models.MovimientoDetalle{}, outputError
+	} else {
+		logs.Debug(result)
+		var lastMovimientoDetalle models.MovimientoDetalle
+		formatdata.FillStruct(result[0], &lastMovimientoDetalle)
+		return lastMovimientoDetalle, nil
+	}
+}
+
+func GetAllUltimos(cuentasMovimientoDetalle []models.CuentasMovimientoProcesoExterno) (cuentasMovimientoDetalleRespuesta []models.MovimientoDetalle, outputError map[string]interface{}) {
+	defer func() {
+		if err := recover(); err != nil {
+			outputError = map[string]interface{}{
+				"funcion": "GetAllUltimos - Unhandled Error!",
+				"err":     err,
+				"status":  "500",
+			}
+			panic(outputError)
+		}
+	}()
+
+	for i := range cuentasMovimientoDetalle {
+		if resultado, err := GetUltimo(cuentasMovimientoDetalle[i]); err != nil {
+			logs.Error(err)
+			outputError = map[string]interface{}{
+				"funcion": "GetAllUltimos - GetUltimo(cuentasMovimientoDetalle[i])",
+				"err":     err,
+				"status":  "502",
+			}
+			return []models.MovimientoDetalle{}, outputError
```

Comment:
Esto se puede reducir, revisar...
```suggestion
			return []models.MovimientoDetalle{}, err
```